internal/tools/http_method.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package tools
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// HTTPMethod is a string of a valid HTTP method (e.g "GET")
+type HTTPMethod string
+
+func (i *HTTPMethod) String() string {
+	if string(*i) != "" {
+		return strings.ToUpper(string(*i))
+	}
+	return ""
+}
+
+// isValidHTTPMethod checks if the input string matches one of the method constants defined in the net/http package
+func isValidHTTPMethod(method string) bool {
+	switch method {
+	case http.MethodGet, http.MethodPost, http.MethodPut, http.MethodDelete,
+		http.MethodPatch, http.MethodHead, http.MethodOptions, http.MethodTrace,
+		http.MethodConnect:
+		return true
+	}
+	return false
+}
```

Comment:
nit: maybe small readability improvement here:

```suggestion
// Use a map for efficient lookups. The empty struct{} uses zero memory.
var standardHTTPMethods = map[string]string{
    http.MethodGet:     {},
    http.MethodHead:   {},
    http.MethodPost:    {},
    http.MethodPut:     {},
    http.MethodPatch:   {},
    http.MethodDelete:  {},
    http.MethodConnect:{},
    http.MethodOptions:{},
    http.MethodTrace:   {},
}

// isValidHTTPMethodMap checks if the method exists in our predefined map.
func isValidHTTPMethodMap(method string) (string, error) {
    v, ok := standardHTTPMethods[strings.ToUpper(method)]
    if !ok {
        return nil, err
    }
    return v, err
}
```